ChangeLog.d/psa-openless.txt

```diff
@@ -0,0 +1,17 @@
+Features
+   * In the PSA API, it is no longer necessary to open persistent keys:
+     operations now accept the key identifier. The type psa_key_handle_t is now
+     identical to psa_key_id_t instead of being platform-defined. This bridges
+     the last major gap to compliance with the PSA Cryptography specification
+     version 1.0.0. Opening persistent keys is still supported for backward
+     compatibility, but will be deprecated and later removed in future
+     releases.
+
+Bugfix
+   * psa_set_key_id() no longer resets the key location when enforcing the
+     key lifetime to be non-volatile.
```

Comment:
We both got this wrong. `psa_set_key_id` didn't reset the key location (except during an intermediate version of this PR IIRC). The actual change is from
```
if persistence == VOLATILE and location == LOCAL: persistence = PERSISTENT
```
to
```
if persistence == VOLATILE: persistence = PERSISTENT
```

So a correct description would be

> psa_set_key_id() now also sets the lifetime to persistent for keys located in a secure element.

Reply:
Yes after looking at the version of the function in `development` I agree with that. I seems I got eventually lost in the various versions of the function.

Reply:
I've fixed that in the "Add change log" commit and force-pushed the result.